apps/remix-ide/src/blockchain/blockchain.js

```diff
@@ -170,15 +170,39 @@ export class Blockchain extends Plugin {
     }
     const continueCb = (error, continueTxExecution, cancelCb) => { continueTxExecution() }
     const promptCb = (okCb, cancelCb) => { okCb() }
-    const finalCb = (error, txResult, address, returnValue) => {
+    const finalCb = async (error, txResult, address, returnValue) => {
       if (error) {
         const log = logBuilder(error)
   
         _paq.push(['trackEvent', 'blockchain', 'Deploy With Proxy', 'Proxy deployment failed: ' + error])
         return this.call('terminal', 'logHtml', log)
       }
-      if (networkInfo.name === 'VM') this.config.set('vm/proxy', address)
-      else this.config.set(`${networkInfo.name}/${networkInfo.currentFork}/${networkInfo.id}/proxy`, address)
+      const hasPreviousDeploys = await this.call('fileManager', 'exists', `.deploys/upgradeable-contracts/${networkInfo.name}.json`)
+      
+      // make deploys folder read only.
+      if (hasPreviousDeploys) {
+        const deployments = await this.call('fileManager', 'readFile', `.deploys/upgradeable-contracts/${networkInfo.name}.json`)
+        const parsedDeployments = JSON.parse(deployments)
+
+        parsedDeployments.deployments.push({
+          fork: networkInfo.currentFork,
+          proxyAddress: address,
+          layout: implementationContractObject.contract.object.storageLayout,
+          date: new Date().toISOString()
+        })
+        await this.call('fileManager', 'writeFile', `.deploys/upgradeable-contracts/${networkInfo.name}.json`, JSON.stringify(parsedDeployments, null, 2))
+      } else {
+        await this.call('fileManager', 'writeFile', `.deploys/upgradeable-contracts/${networkInfo.name}.json`, JSON.stringify({
```

Comment:
I think we should find a way to not mix the different network.
maybe the solution would be to create a subfolder for each network.

Reply:
Ok, i will make the change.

Reply:
Upgrades don't have anything to do with previous implementation address, so i assumed it will not be needed. I can add it if you like.

Reply:
Current implementation address becomes previous implementation address when upgrade is initiated. Anyway, i will add it.